droidsx/droid-bodies
# publish telekinesis goals from keyboard

## Summary

Unified telekinesis and keyboard control for the Pincher X100 robot, replacing key strings commands with the proper goal messages. Basic goal processing improvements have been included as well.

### What changed?

- Renamed `px100_moveit_interface.launch.py` to `px100_telekinesis.launch.py` and updated the executable to `telekinesis_control_node`.
- Refactored `keyboard_publisher.py` to publish `TelekinesisGoal` messages instead of simple keyboard inputs.
- Updated `CMakeLists.txt` and `package.xml` to include new dependencies and build the `telekinesis_control_node`.
- Replaced `keyboard_control_node.cpp` with `telekinesis_control_node.cpp`, implementing a more sophisticated control system using `TelekinesisGoal` messages.
- Modified the queue policy for the goal subscriber. Now the movements follow the bag duration.

### Why is This Change Required?

This change was prioritized to support debug activities related to goal processing. The improved keyboard control node now allows to diagnose goal processing more easily. The goal processing node also provides a foundation for future enhancements, such as separating arm and gripper requests into different threads.

### How to test?

1. Launch the telekinesis control system using the updated launch file.

```bash
ros2 launch pincher_x100 px100_telekinesis.launch.py hardware_type:=gz_classic
```
2. Run the keyboard publisher node.

```bash
ros2 run pincher_x100 keyboard_publisher
```

3. Use the keyboard controls (w, s, a, d, r, f) to move the robot's end effector. Note: The gripper has been disabled while debug activities are still ongoing.

4. Verify that the robot responds accurately to the telekinesis goals.
